templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml

```diff
@@ -52,14 +52,21 @@ contents:
             "{{ onPremPlatformIngressIP . }}")
         DOMAIN="{{.DNS.Spec.BaseDomain}}"
         if [[ -n "$NAMESERVER_IP" ]]; then
-            >&2 echo "NM resolv-prepender: Prepending 'nameserver $NAMESERVER_IP' to /etc/resolv.conf (other nameservers from /var/run/NetworkManager/resolv.conf)"
-            sed -e "/^search/d" \
-                -e "/Generated by/c# Generated by KNI resolv prepender NM dispatcher script\nsearch $DOMAIN\nnameserver $NAMESERVER_IP" \
-                /var/run/NetworkManager/resolv.conf > /etc/resolv.tmp
+            source /etc/os-release
+            if [[ "$NAME" == "Fedora" ]]; then
+                echo "[Resolve]" > /etc/systemd/resolved.conf.d/60-kni.conf
+                echo "DNS=$NAMESERVER_IP" >> /etc/systemd/resolved.conf.d/60-kni.conf
+                echo "Domains=$DOMAIN" >> /etc/systemd/resolved.conf.d/60-kni.conf
```

Comment:
Wouldn't it be nicer to have a single heredoc or an echo with "\n" instead of three separate echo?

Reply:
The most cleanest way would be creating this file unconditionally - and it would be caught up in RHEL when it does the switch. WDYT?

Reply:
I don't have a preference how to write the file, seems echo with '\n' would the cleanest option. I don't mind rewriting it if you insist